### PR TITLE
RavenDB-14932 verify that rollups can be indexed, added Start/End to dynamic timeseries segment

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Static/TimeSeries/DynamicTimeSeriesSegment.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public override bool Set(object item)
 
         public DynamicArray Sum => _sum ??= new DynamicArray(Summary.Sum);
 
-        public int Count => Summary.Count;
+        public dynamic Count => Summary.Count;
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Static/TimeSeries/DynamicTimeSeriesSegment.cs]: Why?

File: src/Raven.Server/Documents/Indexes/Static/TimeSeries/DynamicTimeSeriesSegment.cs [hunk 2]
```diff
@@ -73,7 +73,9 @@ public LazyStringValue Name
             }
         }
 
-        public dynamic Baseline => TypeConverter.ToDynamicType(_segmentEntry.Baseline);
+        public dynamic Start => TypeConverter.ToDynamicType(_segmentEntry.Start);
+
+        public dynamic End => TypeConverter.ToDynamicType(_segmentEntry.Segment.GetLastTimestamp(_segmentEntry.Start));
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Static/TimeSeries/DynamicTimeSeriesSegment.cs]: `ToDynamicType` does nothing here, returns a `DateTime`, why not just use that?

**Comment by ppekrol** [comment]
Need to check if this is also required for counters.

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 3]
```diff
@@ -2040,22 +2040,48 @@ private static TimeSeriesSegmentEntry CreateTimeSeriesItem(DocumentsOperationCon
...
                 SegmentSize = segmentSize,
                 Collection = DocumentsStorage.TableValueToId(context, (int)TimeSeriesTable.Collection, ref reader),
-                Baseline = baseline,
+                Start = baseline,
                 Etag = Bits.SwapBytes(etag),
             };
+
+            static LazyStringValue ToLuceneKey(DocumentsOperationContext context, LazyStringValue documentId, LazyStringValue name, DateTime baseline)
+            {
+                var ticksSpan = Encoding.UTF8.GetBytes(baseline.Ticks.ToString()).AsSpan();
```
Comment (ppekrol) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:2063]: @ayende Can we handle that better? Any ideas?

Comment (ayende) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:2063]: Any reason not to use this?
https://docs.microsoft.com/en-us/dotnet/api/system.buffers.text.utf8formatter.tryformat?view=netcore-3.1#System_Buffers_Text_Utf8Formatter_TryFormat_System_Int64_System_Span_System_Byte__System_Int32__System_Buffers_StandardFormat_

**Pull request merged**